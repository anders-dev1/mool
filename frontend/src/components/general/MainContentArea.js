import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

const ContentContainer = styled.div`
  // This container limits the content to a smaller area in the center of the viewport.
  width: calc(100% - 1em);
  height: 100%;

  @media (min-width: 60em) {
    width: 58em;
  }
`;

class MainContentArea extends React.Component {
    render() {
        return (
            <Container>
                <ContentContainer>
                    {this.props.children}
                </ContentContainer>
            </Container>
        );
    }
}

export default MainContentArea;