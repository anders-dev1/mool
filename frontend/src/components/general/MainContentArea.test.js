import {render} from "@testing-library/react";
import MainContentArea from "./MainContentArea";

test('renders correctly', () => {
    const {container} = render(
        <MainContentArea><p>This content should be inside the area :)</p></MainContentArea>
    );

    expect(container).toMatchSnapshot();
});