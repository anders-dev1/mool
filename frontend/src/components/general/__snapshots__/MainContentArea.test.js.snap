// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ fEUKnN"
  >
    <div
      class="sc-gsnTZi bvivOo"
    >
      <p>
        This content should be inside the area :)
      </p>
    </div>
  </div>
</div>
`;
